src/main/java/com/microsoft/graph/content/BatchResponseContent.java

```diff
@@ -0,0 +1,228 @@
+package com.microsoft.graph.content;
+
+import com.google.common.io.ByteStreams;
+import com.google.gson.*;
+import com.microsoft.graph.CoreConstants;
+import com.microsoft.graph.exceptions.ErrorConstants;
+import com.microsoft.graph.requests.ResponseBodyHandler;
+import com.microsoft.kiota.ResponseHandler;
+import com.microsoft.kiota.serialization.Parsable;
+import com.microsoft.kiota.serialization.ParsableFactory;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.*;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A class representing the content of a batch request response.
+ */
+public class BatchResponseContent {
+    private Response batchResponse;
+    private JsonObject jsonBatchResponseObject;
+    private HashMap<String, ParsableFactory<? extends Parsable>> apiErrorMappings;
+    private Protocol batchResponseProtocol;
+    private Request batchResponseRequest;
+    /**
+     * Creates a new BatchResponseContent instance.
+     * @param batchResponse The response of the batch request.
+     */
+    public BatchResponseContent(@Nonnull Response batchResponse) {
+        this(batchResponse, null);
+    }
+    /**
+     * Creates a new BatchResponseContent instance.
+     * @param batchResponse The response of the batch request.
+     * @param apiErrorMappings The error mappings to use when deserializing failed responses bodies. Where an error code like 401 applies specifically to that status code, a class code like 4XX applies to all status codes within the range if the specific error code is not present.
+     */
+    public BatchResponseContent(@Nonnull Response batchResponse, @Nullable Map<String, ParsableFactory<? extends Parsable>> apiErrorMappings) {
+        this.batchResponse = Objects.requireNonNull(batchResponse, ErrorConstants.Messages.NULL_PARAMETER + "batchResponse");
```

Comment:
`Global comment`
I notice the `ErrorConstants.Messages.NULL_PARAMETER` is defined as a string template, however, we are using concatenation all over the code. Will this still result in string interpolation?

Reply:
Yes, I purposely made the Null_Parameter message the following
"The following parameter cannot be null: "; 
At which point we just concatenate the param name at the end. There were code smells which complained about using String.format() for a template where only a single parameter was being used. It suggested concatenation for performance so I adjusted the message so that it would not require String.format().